log/sequencer.go

```diff
@@ -290,9 +293,14 @@ func (s Sequencer) IntegrateBatch(ctx context.Context, logID int64, limit int, g
 	start := s.timeSource.Now()
 	label := strconv.FormatInt(logID, 10)
 
+	tree, err := trees.GetTree(ctx, s.adminStorage, logID, seqOpts)
```

Comment:
`Sequencer.ExecutePass` already has a tree for the given `logID`, it should probably pass that in to `ItegrateBatch` (perhaps instead of `logID`?)

Then `Sequencer` wouldn't need an `AdminStorage` either, which looks very unusual!

Reply:
It's still somewhat WIP while I fix everything up. I'll look at this as well. I was probably overzealous in making everything go through GetTree because there were RPC server paths that didn't and relied on the callback through storage.